pulse2percept/models/beyeler2019.py

```diff
@@ -105,6 +106,27 @@ def _predict_spatial(self, earray, stim):
                                self.thresh_percept,
                                self.n_threads)
 
+    def __deepcopy__(self, memodict={}):
+        """
+        Perform a deep copy of the ScoreboardSpatial object.
+        Parameters
+        ----------
+        memodict: dict
+            Dictionary of objects already copied during the current copying pass.
+
+        Returns
+        -------
+            Deep copy of the object
+        """
+        if id(self) in memodict:
+            return memodict[id(self)]
+        attributes = copy.deepcopy(self.__dict__)
+        # Remove attributes set internally
+        attributes.pop('_is_built')
```

Comment:
Why do we not set `_is_built ` and `grid` again after popping them from attributes? Will this result in a built scoreboard model not being built after copying?

Reply:
I think this would be solved by using the parent `BaseModel` `deepcopy` which copies all attributes manually after creation

Reply:
This is a good point. Thank you for the feedback, I will work on integrating this feedback, as well as the other feedback.